ChiModules/KEigenvalueSolver/initialize_k_solver.cc

```diff
@@ -27,17 +29,22 @@ void KEigenvalue::Solver::InitializeKSolver()
      //       precursor IDs 0-5 and material 1 will receive
      //       precursor IDs 6-11.
     **/
-    num_precursors = 0;
+    precursor_map.clear();
+
+    // Loop over materials
     for (int x = 0; x < material_xs.size(); x++) {
```

Comment:
Clang-tidy tells me this loop can be range based. Please change to the suggested below:

```c++
for (auto & xs : material_xs) {
  // Material precursor mapping vector
  std::vector<size_t> mat_map;

  // Define the precursor mapping for this material
  for (int j = 0; j < xs->num_precursors; ++j)
     mat_map.emplace_back(num_precursors + j);

  // Increment the total number of precursors
  num_precursors += xs->num_precursors;

  // Add mapping to the precursor map
  precursor_map.emplace_back(mat_map);
}
```

Reply:
Done.